source/js/ui-scripts.js

```diff
@@ -72,239 +85,325 @@ function hideLoader() {
+	$('#' + divId).show();
+}
+function hideError(divId) {
+	$('#' + divId).html('msg');
+	$('#' + divId).hide();
+}
+function showLoader() {
+	$("*[data-loader='y']").show();
+}
+function hideLoader() {
+	$("*[data-loader='y']").hide();
+}
+
+function daysInMonth(month, year) {
+	return new Date(year, month, 0).getDate();
+}
+
+function showErrorAlert(container, text) {
+	resetAlert(container);
+	container.html(text);
+	container.addClass('alert alert-danger');
+	container.show();
+}
 
-  function showWarningAlert(container, text) {
-   resetAlert(container);
-   container.html(text);
-   container.addClass("alert alert-warning");
-   container.show();
-  }
+function showWarningAlert(container, text) {
+	resetAlert(container);
+	container.html(text);
+	container.addClass('alert alert-warning');
+	container.show();
+}
 
-  function showInfoAlert(container, text) {
-   resetAlert(container);
-   container.html(text);
-   container.addClass("alert alert-info");
-   container.show();
-  }
-  function showSuccessAlert(container, text) {
-   resetAlert(container);
-   container.html(text);
-   container.addClass("alert alert-success");
-   container.show();
-  }
-  function resetAlert(container) {
-   container.html("text");
-   container.attr('class','');
-   container.hide();
-  }
-  function fadeInAndOut(element) {
-	element.fadeTo( "slow" , 0.25, function() {
-		element.fadeTo( "slow" , 1.0, function() {
-			fadeInAndOut(element)
-		  });
+function showInfoAlert(container, text) {
+	resetAlert(container);
+	container.html(text);
+	container.addClass('alert alert-info');
+	container.show();
+}
+function showSuccessAlert(container, text) {
+	resetAlert(container);
+	container.html(text);
+	container.addClass('alert alert-success');
+	container.show();
+}
+function resetAlert(container) {
+	container.html('text');
+	container.attr('class', '');
+	container.hide();
+}
+function fadeInAndOut(element) {
+	element.fadeTo('slow', 0.25, function () {
+		element.fadeTo('slow', 1.0, function () {
+			fadeInAndOut(element);
+		});
 	});
 }
 
-  //toggleCheckbox = checkbox when click on table row
-  $.fn.toggleCheckbox = function() {
-	  this.prop('checked', !this.prop('checked'));
-  }
-  $(function ()
-  {
-	  $('.toggleCheckbox tr').click(function() {
-			  $(this).find('td input:checkbox').toggleCheckbox();
-	  })
-	  $('.toggleCheckbox tr td input:checkbox').click(function(event) {
-			  event.stopPropagation();
-	  })
-  });
+var groupMockHtml = " <style>#group_span fieldset{margin-top: 15px;}#group_span fieldset legend{width: auto; margin-bottom: 0; border-bottom: 0; font-size: 1.2em;}#group_span fieldset b{font-weight: normal;}</style> <fieldset> <legend> Automatic selection </legend> <b>One specialist will be selected automatically from:</b><br/> &nbsp&nbsp&nbsp&nbsp<b>Dr Roland Rat</b><br/> &nbsp&nbsp&nbsp&nbsp<b>Professor Wise Old Owl</b><br/> &nbsp&nbsp&nbsp&nbsp<b>Mr Fred Flintstone</b><br/> </fieldset>"
 
+function validateGroup(){
```

Comment:
Leave this function here, but don't modify it with mock stuff.

Instead, intercept the onClick function using jQuery and execute the mock when running JS inside the patternlib